doc: Fix typos and grammar in ecbackend.rst (#42365)
ceph/ceph: doc/dev/osd_internals/erasure_coding/ecbackend.rst

----------------------------------

With the replicated strategy, all replicas of a PG are
interchangeable.  With erasure-coding, different positions in the
acting set have different pieces of the erasure-coding scheme and are
not interchangeable.  Worse, crush might cause chunk 2 to be written
to an OSD which happens already to contain an (old) copy of chunk 4.
This means that the OSD and PG messages need to work in terms of a

> type like pair<shard_t, pg_t> in order to distinguish different pg

Reviewer: we can also uppercase pg here

----------------------------------

With the replicated strategy, all replicas of a PG are

> interchangeable.  With erasure-coding, different positions in the

Reviewer 2: i don't think "erasure-coding" is better than "erasure coding" here. as it is a noun not an adjective in this context. the same applies to "objectstore" versus "object-store".

Author: Changed all the occurrences

OSD Object Write and Consistency
--------------------------------

Regardless of the algorithm chosen above, writing of the data is a two-
phase process: commit and rollforward. The primary sends the log
entries with the operation described (see
osd_types.h:TransactionInfo::(LocalRollForward|LocalRollBack).  
In all cases, the "commit" is performed in place, possibly leaving some
information required for a rollback in a write-aside object.  The

> rollforward phase occurs once all acting set replicas have committed

Reviewer: rollforward phase occurs once all acting set replicas have finalized(committed)
```suggestion
rollforward phase occurs once all acting set replicas have finalized(committed)
```

> the commit (sorry, overloaded term) and remove the rollback information.

Reviewer: we can remove `(sorry, overloaded term)` here 
```suggestion
the commit, it then removes the rollback information.
```
this is a follow-up operation, the reader might misunderstand not understand it's interdependence with this change

> rollforward phase occurs once all acting set replicas have finalized(committed)

Reviewer 2: i'd prefer using "commit" instead of "finalize" even we are using this word as both verb and a noun in the same sentence. i think we are used to put it on daily communication.

Author: Changed

See ExtentCache.h for a detailed explanation of how the cache
states correspond to the higher level invariants about the conditions
under which concurrent operations can refer to the same object.

Pipeline
--------

Reading src/osd/ExtentCache.h should have given a good idea of how
operations might overlap.  There are several states involved in
processing a write operation and an important invariant which
isn't enforced by PrimaryLogPG at a higher level which needs to be
managed by ECBackend.  The important invariant is that we can't

> have uncacheable and rmw operations running at the same time

Reviewer: can you also fix `s/rmw/read–modify–write`  rmw might not be clear to beginner

Author: Sure!
